From a33c262f828f803fffdad8e1f44d524dc9c75856 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 3 Aug 2022 01:10:01 +0200
Subject: [PATCH] fcntl: Fix build failure for some architectures with Linux
 5.19

Starting from Linux 5.19, the kernel UAPI headers now only define
__ARCH_FLOCK64_PAD if the architecture actually needs padding in
struct flock64.  Wrap its use with #ifdef,

Upstream-Status: Backport [https://git.kernel.org/pub/scm/libs/klibc/klibc.git/commit/?id=bb2fde5ddbc18a2e7795ca4d24759230c2aae9d0]
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 usr/include/fcntl.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/include/fcntl.h b/usr/include/fcntl.h
index ed703a6..cb2e4e5 100644
--- a/usr/include/fcntl.h
+++ b/usr/include/fcntl.h
@@ -33,7 +33,9 @@ struct flock {
 	__kernel_loff_t l_start;
 	__kernel_loff_t l_len;
 	__kernel_pid_t  l_pid;
+#ifdef __ARCH_FLOCK64_PAD
         __ARCH_FLOCK64_PAD
+#endif
 };
 
 #ifdef F_GETLK64
-- 
2.37.2

